hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClusterListenerSupport.java

```diff
@@ -233,7 +233,16 @@ public void run() {
                             connectToCluster();
                         } catch (Exception e) {
                             logger.warning("Could not re-connect to cluster shutting down the client", e);
-                            client.getLifecycleService().shutdown();
+                            new Thread(new Runnable() {
```

Comment:
No, it is not possible. If we do that we end up with same problem but with different executor.
When `client.getLifecycleService().shutdown()`  returns, it has to make sure all executors are closed. We should not call `client.getLifecycleService().shutdown()`  inside a thread that it needs to close. 
Similar usage on server side 
https://github.com/hazelcast/hazelcast/pull/9508/files#diff-b84f3952c1015997c1cb46310f7af2a1R43